playbooks/provisioning/openstack/server-setup.yaml

```diff
@@ -0,0 +1,72 @@
+---
+- hosts: cluster_hosts
+  become: true
+  pre_tasks:
+  - include: pre_tasks.yml
+  roles:
+  - role: hostnames
+
+# Subscribe Hosts
+- hosts: cluster_hosts
+  become: true
+  roles:
+  - role: subscription-manager
```

Comment:
this is probably incredibly premature request but it may be worth looking into using the rhel_subscribe role

/usr/share/ansible/openshift-ansible/roles/rhel_subscribe/